CI issues, fairchem and scipy

There are some issues causing the CI to fail https://github.com/stfc/janus-core/actions/runs/20988865093/job/60329171098?pr=653

First scipy [1.17.0](
https://github.com/scipy/scipy/releases/tag/v1.17.0) has removed ```sph_harm```, which is being imported by something in the equiformer tests. It is not immediately clear where.

```
FAILED tests/test_mlip_calculators.py::test_mlips[equiformer-cpu-kwargs10] - ImportError: cannot import name 'sph_harm' from 'scipy.special' (/home/runner/work/janus-core/janus-core/.venv/lib/python3.12/site-packages/scipy/special/__init__.py)
```

Second, maybe related, something is happening with fairchem and imports/paths

```
FAILED tests/test_mlip_calculators.py::test_mlips[equiformer-cpu-kwargs11] - RuntimeError: Failed to find the trainer 'equiformerv2_forces'. You may either use a trainer from the registry (one of 'base', 'forces', 'energy' or 'ocp') or provide the full import path to the trainer (e.g., 'fairchem.core.trainers.ocp_trainer.OCPTrainer').
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -96,7 +96,6 @@ sevennet = [
 ]
 fairchem-2 = [
     "fairchem-core == 2.12.0",
-    "scipy < 1.17.0",
 ]
 
 # MLIPs with dgl dependency

